### PR TITLE
Add Hotjar heatmap tracking code to page header

### PR DESCRIPTION
**Pull request by dju90**
This adds code to the page header (only on the teacher homepage) needed for Hotjar heatmap tracking analytics. Modeled after #55778, but without the privacy concern associated with Microsoft Clarity.

## Links
- Jira: [TEACH-867](https://codedotorg.atlassian.net/browse/TEACH-867)

## Testing story
Tested locally, supplied function shows up in the header as requested.
_As a teacher_
image
image

_As a student_
image

## Privacy
Cleared provider with legal.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/layouts/_hotjar.html.haml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-# Code snippet needed for Hotjar heatmap tracking.
```
Comment (lfryemason) [inline dashboard/app/views/layouts/_hotjar.html.haml:1]: Could you link to the documentation for this added code?

**Review by lfryemason — COMMENTED** [review]
Looks good, just want to confirm that we have permission from legal and eng leads to add this?

File: dashboard/app/views/layouts/_hotjar.html.haml (hunk 1 above)
Comment (dju90) [reply dashboard/app/views/layouts/_hotjar.html.haml:1]: https://help.hotjar.com/hc/en-us/articles/115011639927-What-is-the-Hotjar-Tracking-Code

**Comment by dju90** [comment]
> Looks good, just want to confirm that we have permission from legal and eng leads to add this?

Permission from legal yes; will talk to Vijaya about it this afternoon.

**Review by lfryemason — APPROVED** [review]

**Review by unlox775-code-dot-org — APPROVED** [review]
Looks great! Thank you for including me in the review!

**Pull request merged**